website/content/docs/concepts/plugins/storage/host-volumes.mdx

```diff
@@ -0,0 +1,387 @@
+the plugin on the Nomad node for volume scheduling.
+
+**CLI arguments:** `$1=fingerprint`
+
+**Environment variables:**
+
+```
+DHV_OPERATION=fingerprint
+```
+
+**Expected stdout:**
+
+```
+{"version": "0.0.1"}
+```
+
+**Requirements:**
+
+* Must complete within 5 seconds, or Nomad will kill it
+  * should be much faster, as no actual work should be done
+* "version" value must be valid per the [hashicorp/go-version][go-version]
+  golang package
+
+</blockquote>
+
+#### create
+<blockquote style={{borderLeft: "solid 1px #00ca8e"}}>
+
+Nomad calls `create` when you run [`nomad volume create`][cli-create] CLI or
+use the [create API][api-create]).
+
+You can run `create` again for the same volume if you include the volume
+`id` in the volume specification.
+
+Nomad also calls `create` when the agent starts, for each volume that was
+previously created on the node, so plugins can ensure the volumes are available
+after an agent restart or host reboot.
+
+**CLI Arguments:** `$1=create`
+
+**Environment variables:**
+
+```
+DHV_OPERATION=create
+DHV_HOST_PATH=/suggested/path/to/place/volume
+DHV_NODE_ID={Nomad node ID}
+DHV_NODE_POOL={Nomad node pool}
+DHV_NAMESPACE={volume namespace}
+DHV_VOLUME_NAME={name from the volume specification}
+DHV_VOLUME_ID={volume ID assigned by Nomad}
+DHV_CAPACITY_MIN_BYTES={capacity_min from the volume spec}
+DHV_CAPACITY_MAX_BYTES={capacity_max from the volume spec}
+DHV_PARAMETERS={json of parameters from the volume spec, or: null}
+DHV_PLUGIN_DIR={path to directory containing plugins}
+```
+
+**Expected stdout:**
+
+```
+{"path": "/path/to/created/volume", "bytes": 50000000}
+```
+
+**Requirements:**
+
+* Must complete within 60 seconds, or Nomad will kill it
+* Must be idempotent - repeat runs produce the same result
+* Must be safe to run concurrently, per volume name per node
+* Returned "path" must be the location of the created volume.
+  Nomad will mount this path into workloads that request the volume.
+
+</blockquote>
+
+#### delete
+<blockquote style={{borderLeft: "solid 1px #00ca8e"}}>
+
+Nomad calls `delete` when you run [`nomad volume delete`][cli-delete] CLI or
+use the [delete API][api-delete]).
+
+Nomad also runs `delete` when an initial volume `create` operation
+fails, since it may have been only partially completed.
+
+**CLI Arguments:** `$1=delete`
+
+**Environment variables:**
+
+```
+DHV_OPERATION=delete
+DHV_HOST_PATH=/suggested/path/to/place/volumes
+DHV_NODE_ID={Nomad node ID}
+DHV_NODE_POOL={Nomad node pool}
+DHV_NAMESPACE={volume namespace}
+DHV_VOLUME_NAME={name from the volume specification}
+DHV_VOLUME_ID={volume ID assigned by Nomad}
+DHV_PARAMETERS={json of parameters from the volume spec, or: null}
+DHV_PLUGIN_DIR={path to directory containing plugins}
+```
+
+**Expected stdout:** none (stdout is discarded)
+
+**Requirements:**
+
+* Must complete within 60 seconds, or Nomad will kill it
+* Must be idempotent - repeat runs produce the same result
```

Comment:
It's implied by idempotency, but we should probably refine this further to explicitly say that calling delete on an already deleted volume must not return an error. Just so there are no doubts for authors.